agimus_controller/agimus_controller/trajectories/generic_visual_servoing_trajectory.py

```diff
@@ -0,0 +1,110 @@
+        self.init_in_world_M_object = None
+
+        # if distance to goal is below this threshold, start visual servoing
+        self.start_visual_servoing_time_threshold = (
+            traj_params.start_visual_servoing_time_threshold
+        )
+
+        # another attribute to decide when to activate visual servoing by giving a trajectory index
+        # if set, it has the priority over attribute start_visual_servoing_time_threshold
+        self.activate_visual_servoing_idx = None
+
+        # boolean to decides to activate visual servoing
+        self.activate_visual_servoing = False
+
+        # Wether current trajectory requires visual servoing or not
+        self.use_visual_servoing = False
+
+    def update_activation_of_visual_servoing(self):
+        """Update boolean regarding activation of visual servoing."""
+        if self.use_visual_servoing:
+            if self.activate_visual_servoing_idx is not None:
+                self.activate_visual_servoing = (
+                    self.traj_idx >= self.activate_visual_servoing_idx
+                )
+            else:
+                time_to_reach_goal = (
+                    len(self.trajectory) - 1 - self.traj_idx
+                ) * self.dt
+                self.activate_visual_servoing = (
+                    time_to_reach_goal < self.start_visual_servoing_time_threshold
+                )
+        else:
+            self.activate_visual_servoing = False
+        if not self.activate_visual_servoing:
+            self.visual_servoing_time = 0.0
+
+    def add_trajectory(
+        self,
+        trajectory,
+        use_visual_servoing,
+        init_in_world_M_object=None,
+        activate_visual_servoing_idx=None,
+    ):
+        if use_visual_servoing:
+            if init_in_world_M_object is None:
+                raise ValueError("Init pose detection not set.")
+            self.init_in_world_M_object = pin.XYZQUATToSE3(init_in_world_M_object)
+            self.activate_visual_servoing_idx = activate_visual_servoing_idx
+        else:
+            self.init_in_world_M_object = None
+        super().add_trajectory(trajectory)
+        self.use_visual_servoing = use_visual_servoing
+
+    def get_traj_point_at_t(self, t: np.float64) -> WeightedTrajectoryPoint:
+        self.update_activation_of_visual_servoing()
+        traj_point = self.trajectory[self.traj_idx]
+        key = next(iter(traj_point.end_effector_poses))
+        if self.init_in_world_M_object is not None:
+            in_world_M_ee = pin.XYZQUATToSE3(traj_point.end_effector_poses[key])
+            in_object_M_ee = self.init_in_world_M_object.inverse() * in_world_M_ee
+            traj_point.end_effector_poses[key] = pin.SE3ToXYZQUAT(in_object_M_ee)
+        if self.activate_visual_servoing:
+            self.w_pose = [
+                self.w_increasing.get_weight_at_t(self.visual_servoing_time)
+            ] * 6
+            self.visual_servoing_time += 0.01
```

Comment:
To avoid hardcoding the timestep:
```suggestion
            assert t >= self.visual_servoing_time
            self.w_pose = [
                self.w_increasing.get_weight_at_t(t - self.visual_servoing_time)
            ] * 6
```